### PR TITLE
Closes #78 - Allow multiple LdapContextSource beans

### LINKED ISSUE #78: Allow more than one LdapContextSource bean to be created from Spring Boot applications
Spring Boot applications that use kadai-rest-spring are not allowed to create more than one Bean of Type LdapContextSource.

It should be allowed to have 2 or more beans of type LdapContextSource, in case the applications use more than one LDAP in their code.

### PR DESCRIPTION
**Pull request by CRoberto1926**
### Thanks for your PR! Please fill out the following list :)

---

- [ ] I put the ticket or multiple tickets in review
- [x] Commit message format → Closes #&lt;Issue Number&gt; - Your commit message.
- [x] Sonarcloud link : https://sonarcloud.io/summary/new_code?id=CRoberto1926_kadai&branch=feature%2FKAD-62-multiple-context-source-beans
- [ ] No documentation update needed
- [x] Link to PR with documentation update: https://github.com/kadai-io/kadai-doc/pull/14
- [ ] No Release Notes needed
- [x] Release Notes :

```
- Allow more than one LdapContextSource bean to be created
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -77,3 +77,4 @@ Thumbs.db
 
 # DB2 database
 db.tar.gz
+.codiumai
```
Comment (holgerhagen) [inline .gitignore:80]: what is this good for?

File: rest/kadai-rest-spring-example-boot/src/main/java/io/kadai/example/boot/security/BootWebSecurityConfigurer.java [hunk 2]
```diff
@@ -170,8 +153,8 @@ public LdapAuthoritiesPopulator authoritiesPopulator(
     return populator;
   }
 
-  @Bean
-  public DefaultSpringSecurityContextSource defaultSpringSecurityContextSource() {
+  @Bean(name = KADAI_LDAP_CONTEXT_SOURCE)
+  public BaseLdapPathContextSource ldapContextSource() {
```
Comment (holgerhagen) [inline rest/kadai-rest-spring-example-boot/src/main/java/io/kadai/example/boot/security/BootWebSecurityConfigurer.java:157]: Why BaseLdap "PATH" ContextSource? Is path a important detail here?

**Review by holgerhagen — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (CRoberto1926) [reply .gitignore:80]: I installed a GenAI plugin for IntelliJ Idea to test it, and it generated a JSON File with the Vector Embeddings in that folder. The plugin also added that line to the .gitignore file. I don't think it makes any harm

File: rest/kadai-rest-spring-example-boot/src/main/java/io/kadai/example/boot/security/BootWebSecurityConfigurer.java (hunk 2 above)
Comment (CRoberto1926) [reply rest/kadai-rest-spring-example-boot/src/main/java/io/kadai/example/boot/security/BootWebSecurityConfigurer.java:157]: BaseLdapPathContextSource is the most general interface that can be created and injected as bean in BootWebSecurityConfigurer. It is just called in this way

**Review by holgerhagen — APPROVED** [review]

**Pull request merged**